env/fs_posix.cc

```diff
@@ -928,11 +937,19 @@ size_t PosixFileSystem::GetLogicalBlockSize(const std::string& fname, int fd) {
 #ifdef OS_LINUX
   return logical_block_size_cache_.GetLogicalBlockSize(fname, fd);
 #else
-  (void) fname;
+  (void)fname;
   return PosixHelper::GetLogicalBlockSizeOfFd(fd);
 #endif
 }
 
+size_t PosixFileSystem::GetLogicalBlockSizeIfNeeded(const EnvOptions& options,
+                                                    const std::string& fname,
+                                                    int fd) {
+  return (options.use_direct_reads || options.use_direct_writes)
```

Comment:
Some users only turn on write or read, so it may makes sense to separate the if for read and write, based on the class is for read or write.

Reply:
Separated out two methods: GetLogicalBlockSizeForReadIfNeeded and GetLogicalBlockSizeForWriteIfNeeded